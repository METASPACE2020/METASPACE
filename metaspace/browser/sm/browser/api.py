import io
import functools

import PIL.Image
import numpy as np
from fastapi import FastAPI
from pydantic import BaseModel
from starlette.responses import StreamingResponse
import uvicorn

from sm.browser import utils
from sm.browser.main import preprocess_dataset_peaks, DatasetBrowser

app = FastAPI()


@functools.lru_cache(maxsize=128)
def load_dataset_browser(s3_path: str):
    return DatasetBrowser(s3_path)


class DatasetPreprocessItem(BaseModel):
    s3_path: str


@app.post("/preprocess")
async def preprocess(item: DatasetPreprocessItem):
    preprocess_dataset_peaks(item.s3_path)
    return {"status": "ok"}


class MzSearchItem(BaseModel):
    s3_path: str
    mz: float
    ppm: int = 3


class PngStreamingResponse(StreamingResponse):
    media_type = "image/png"


@app.post("/search", response_class=PngStreamingResponse)
async def perform_search(item: MzSearchItem):
    dataset_browser = load_dataset_browser(item.s3_path)
    mz_lo, mz_hi = utils.mz_ppm_bin(mz=item.mz, ppm=item.ppm)
    rgba_array = dataset_browser.search(mz_lo, mz_hi)

    image = PIL.Image.fromarray((rgba_array * 255).astype(np.uint8), mode="RGBA")
    fp = io.BytesIO()
    image.save(fp, format="PNG")
    fp.seek(0)
    return PngStreamingResponse(fp)


if __name__ == "__main__":
    uvicorn.run(app)
