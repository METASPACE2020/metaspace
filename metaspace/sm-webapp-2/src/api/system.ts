import gql from 'graphql-tag'

export interface SystemHealth {
  canMutate: boolean
  canProcessDatasets: boolean
  message?: string
}

// Always use fetchPolicy: 'cache-first' for this
export const getSystemHealthQuery = gql`
  query GetSystemHealth {
    systemHealth {
      canMutate
      canProcessDatasets
      message
    }
  }
`

// Always use fetchPolicy: 'cache-first' for this
export const getSystemHealthSubscribeToMore = {
  document: gql`
    subscription SystemHealth {
      systemHealthUpdated {
        canMutate
        canProcessDatasets
        message
      }
    }
  `,
  updateQuery(previousResult: any, { subscriptionData }: any) {
    return {
      systemHealth: subscriptionData.data.systemHealthUpdated,
    }
  },
}

export const updateSystemHealthMutation = gql`
  mutation UpdateSystemHealth($health: UpdateSystemHealthInput!) {
    updateSystemHealth(health: $health)
  }
`

export const getDatabaseOptionsQuery = gql`
  query DatabaseOptions {
    allMolecularDBs {
      id
      name
      version
      archived
      group {
        id
        shortName
      }
    }
  }
`
