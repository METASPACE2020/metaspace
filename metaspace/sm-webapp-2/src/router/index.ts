import { createRouter, createWebHistory } from 'vue-router'
import { Component } from 'vue'
import NotFound from '../modules/App/NotFoundPage.vue'
import AboutPage from '../modules/App/AboutPage'
import DatasetsPage from '../modules/Datasets/DatasetsPage.vue'

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ '../modules/Annotations/AnnotationsPage.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ '../modules/MetadataEditor/MetadataEditPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ '../modules/MetadataEditor/UploadPage.vue'),
  DatasetOverviewPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ '../modules/Datasets/overview/DatasetOverviewPage'),
  DatasetComparisonPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ '../modules/Datasets/comparison/DatasetComparisonPage'),
  DatasetBrowserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ '../modules/Datasets/imzml/DatasetBrowserPage'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/App/HelpPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/Group/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/Project/ProjectsListPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/Group/GroupsListPage'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ '../modules/App/PrivacyPage.vue'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ '../modules/Group/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ '../modules/Project/ViewProjectPage.vue'),

}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, Component>

export const routes : any =[
  { path: '/', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
  { path: '/about', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
  { path: '/annotations', name: 'annotations', component: asyncPages.AnnotationsPage },
  {
    path: '/datasets',
    component: DatasetsPage,
    children: [
      { path: '', component: asyncPages.DatasetTable },
      { path: 'summary', component: asyncPages.DatasetTable },
    ],
  },
  {
    path: '/datasets/:dataset_id/comparison',
    name: 'datasets-comparison',
    component: asyncPages.DatasetComparisonPage,
  },
  { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
  { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.DatasetTable },
  { path: '/dataset/:dataset_id', name: 'dataset-overview', component: asyncPages.DatasetOverviewPage },
  { path: '/dataset/:dataset_id/annotations', name: 'dataset-annotations', component: asyncPages.AnnotationsPage },
  { path: '/dataset/:dataset_id/browser', name: 'dataset-browser', component: asyncPages.DatasetBrowserPage },
  { path: '/dataset/:dataset_id/enrichment', name: 'dataset-enrichment', component: asyncPages.DatasetTable },
  { path: '/upload', component: asyncPages.UploadPage },
  { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },



  { path: '/groups', component: asyncPages.GroupsListPage },
  { path: '/group/create', component: asyncPages.CreateGroupPage },
  { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },


  { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
  { // Legacy URL sent in "request access" emails up until Feb 2019
    path: '/project/:projectIdOrSlug/manage',
    redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } },
  },
  { path: '/projects', component: asyncPages.ProjectsListPage },



  { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },


  {
    path: '/:pathMatch(.*)*',
    component: NotFound,
  },
]

const router = createRouter({
  // @ts-ignore
  history: createWebHistory(import.meta.env.BASE_URL),
  routes,
  scrollBehavior(to, from, savedPosition) {
    // return desired position
    if (savedPosition) {
      return savedPosition;
    }
    return { top: 0 };
  },
})

const pageLoadedAt = Date.now()
router.beforeEach((to, from, next) => {
  // If user has had the window open for a long time, it's likely they're using old code. This can cause issues
  // such as incorrect API usage, or causing error reports for bugs that have already been fixed.
  // Force the browser to reload the page, preferring to do so during page transitions to minimize impact.
  const daysSincePageLoad = (Date.now() - pageLoadedAt) / 1000 / 60 / 60 / 24
  if (to.path !== from.path && daysSincePageLoad > 1.75 || daysSincePageLoad > 7) {
    window.location.assign(to.fullPath)
    next(false)
  } else {
    next()
  }
})


const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
