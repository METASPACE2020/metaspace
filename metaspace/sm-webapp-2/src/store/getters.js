export default {
  filter() {
    return {}
    // return decodeParams(state.route, state.filterLists);
  },

  currentUser(state) {
    return state.currentUser;
  },
}
