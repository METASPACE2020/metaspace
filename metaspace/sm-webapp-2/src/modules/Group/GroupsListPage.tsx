import { computed, defineComponent, reactive } from 'vue'
import { ElButton, ElInput, ElIcon } from '../../lib/element-plus'
import { useQuery } from '@vue/apollo-composable'
import { currentUserRoleQuery, CurrentUserRoleResult } from '../../api/user'
import { getUserGroupsQuery, ViewGroupResult } from '../../api/group'
import { uniqBy } from 'lodash-es'
import GroupsListItem from './GroupsListItem'
import './GroupsListPage.scss'
import { Loading, Search } from '@element-plus/icons-vue'
import { useRouter } from 'vue-router'

interface GroupListPageState {
  groupNameFilter: string | undefined
}

export default defineComponent({
  name: 'GroupsListPage',
  props: {
    className: {
      type: String,
      default: 'groups-list',
    },
  },
  setup: function () {
    const router = useRouter()
    const state = reactive<GroupListPageState>({
      groupNameFilter: '',
    })

    const { result: currentUserResult } = useQuery<CurrentUserRoleResult | any>(currentUserRoleQuery)
    const currentUser = computed(() => (currentUserResult.value != null ? currentUserResult.value.currentUser : null))

    const queryVars = computed(() => ({
      query: state.groupNameFilter,
      useRole: true,
    }))
    const { result: groupsResult, loading: groupsLoading } = useQuery<ViewGroupResult | any>(
      getUserGroupsQuery,
      queryVars,
      { fetchPolicy: 'network-only' }
    )
    const groups = computed(() => (groupsResult.value != null ? groupsResult.value.allGroups : null))

    const handleCreateGroup = async () => {
      await router.push('/group/create')
    }

    return () => {
      if (!currentUser.value && !groupsLoading.value) {
        return (
          <div class="groups-list-container">
            <div class="groups-list-wrapper">
              <p class="font-normal text-center">
                Create an account/login, to create groups and make your work reach more people!
              </p>
            </div>
          </div>
        )
      }

      return (
        <div class="groups-list-container">
          <div class="groups-list-wrapper">
            <div class="groups-list-header">
              <ElInput
                class="group-name-filter tf-outer w-auto"
                modelValue={state.groupNameFilter}
                onInput={(value: string) => {
                  state.groupNameFilter = value
                }}
                placeholder="Enter keywords"
                v-slots={{
                  prepend: () => (
                    <ElIcon class="-mx-1">
                      <Search />
                    </ElIcon>
                  ),
                }}
              />
              <ElButton type="primary" size="large" class="group-list-create-btn" onClick={handleCreateGroup}>
                Create group
              </ElButton>
            </div>
            <div class="groups-list-content">
              {groups.value?.length === 0 && !groupsLoading.value && <p>Group not found!</p>}
              {groups.value?.length > 0 &&
                !groupsLoading.value &&
                uniqBy((groups.value || []) as any[], 'id').map((group: any) => {
                  return (
                    <GroupsListItem
                      id={group.id}
                      name={group.name}
                      shortName={group.shortName}
                      urlSlug={group.urlSlug}
                      currentUserRole={group.currentUserRole}
                      numMembers={group.numMembers}
                    />
                  )
                })}
              {groupsLoading.value && (
                <p>
                  <ElIcon class="is-loading">
                    <Loading />
                  </ElIcon>
                </p>
              )}
            </div>
          </div>
        </div>
      )
    }
  },
})
