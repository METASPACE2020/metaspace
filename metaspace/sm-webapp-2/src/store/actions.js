import apolloClient from '../api/graphqlClient';
import {fetchOptionListsQuery} from '../api/metadata';
import {decodeParams} from '../modules/Filters';
import {computed} from "vue";
import store from "../store/index";
import reportError from "@/lib/reportError";
export default {

  async initFilterLists(context) {
    if(context.state.filterListsLoading || context.state.filterLists != null)
      return;

    context.commit('setFilterListsLoading');
    let response = {}

    try{
      response = await apolloClient.query({
        query: fetchOptionListsQuery,
        fetchPolicy: 'cache-first',
      });
    } catch (err) {
      reportError(err);
    }

    // set annotationIds default according to values passed from snapshot to the store state
    context.commit('setFilterLists', {...response.data,
      annotationIds: computed(() => store.state.snapshotAnnotationIds)} );

    // Refresh the current filter so that computed defaults that depend on `filterLists` are applied
    const filter = decodeParams(context.state.route, context.state.filterLists)
    context.commit('updateFilter', filter)
  },

};
