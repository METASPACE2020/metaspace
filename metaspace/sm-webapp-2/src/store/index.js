import { createStore } from 'vuex'
import mutations from './mutations'
import getters from './getters'
import actions from './actions.js'

import { accountModule } from '../modules/Account'

const store = createStore({
  state: {
    // names of currently shown filters
    orderedActiveFilters: [],

    reportError: false,

    filterLists: null,
    filterListsLoading: false,

    // currently selected annotation
    annotation: undefined,

    // currently selected normalization matrix
    normalization: undefined,

    // roi settings
    roiInfo: { visible: false },

    // is annotation table loading?
    tableIsLoading: true,

    lastUsedFilters: {},

    currentTour: null,

    currentUser: {},

    // ion image global viewer settings
    channels: [],
    mode: 'SINGLE',
    route: {
      path: '',
      params: {},
      query: {},
    },
  },

  getters,
  mutations,
  actions,
  modules: {
    account: accountModule,
  },
})

export default store
