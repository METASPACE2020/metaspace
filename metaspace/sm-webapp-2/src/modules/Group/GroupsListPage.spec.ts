import { defineComponent, nextTick, h, ref } from 'vue'
import { flushPromises, mount } from '@vue/test-utils'
import GroupsListPage from './GroupsListPage'
import { initMockGraphqlClient } from '../../tests/utils/mockGraphqlClient'
import { UserGroupRoleOptions } from '../../api/group'
import { vi } from 'vitest'
import { DefaultApolloClient, useQuery } from '@vue/apollo-composable'
import store from '../../store'
import router from '../../router'
import { encodeParams } from '../Filters'

vi.mock('@vue/apollo-composable', () => ({
  useQuery: vi.fn(),
  useSubscription: vi.fn(() => ({ onResult: vi.fn() })),
  DefaultApolloClient: vi.fn(),
}))

let mockRoutePush

let graphqlMocks: any

describe('GroupsListItem', () => {
  const mockGroups = [
    {
      id: 'group1',
      name: 'group1',
      shortName: 'g1',
      numMembers: 2,
      urlSlug: 'grp1',
      currentUserRole: UserGroupRoleOptions.GROUP_ADMIN,
    },
    {
      id: 'group2',
      name: 'group2',
      shortName: 'g2',
      numMembers: 1,
      urlSlug: 'grp3',
      currentUserRole: UserGroupRoleOptions.MEMBER,
    },
  ]

  const testHarness = defineComponent({
    components: {
      GroupsListPage,
    },
    props: ['className'],
    setup(props) {
      return () => h(GroupsListPage, { ...props })
    },
  })

  const userQuery = async () => {
    const params = {
      currentUser: () => ({ id: 'userid', role: 'user' }),
      allGroups: () => mockGroups,
    }

    graphqlMocks = await initMockGraphqlClient({
      Query: () => params,
    })
    ;(useQuery as any).mockReturnValue({
      result: ref(Object.keys(params).reduce((acc, key) => ({ ...acc, [key]: params[key]() }), {})),
      loading: ref(false),
      onResult: vi.fn(),
    })
  }
  const userNoDataQuery = async () => {
    const params = {
      currentUser: () => ({ id: 'userid', role: 'user' }),
      allGroups: () => [],
    }

    graphqlMocks = await initMockGraphqlClient({
      Query: () => params,
    })
    ;(useQuery as any).mockReturnValue({
      result: ref(Object.keys(params).reduce((acc, key) => ({ ...acc, [key]: params[key]() }), {})),
      loading: ref(false),
      onResult: vi.fn(),
    })
  }

  const noUserQuery = async () => {
    const params = {
      currentUser: () => null,
    }

    graphqlMocks = await initMockGraphqlClient({
      Query: () => params,
    })
    ;(useQuery as any).mockReturnValue({
      result: ref(Object.keys(params).reduce((acc, key) => ({ ...acc, [key]: params[key]() }), {})),
      loading: ref(false),
      onResult: vi.fn(),
    })
  }

  beforeEach(() => {
    mockRoutePush = vi.fn()
    vi.mock('vue-router', async () => {
      const actual: any = await vi.importActual('vue-router')
      return {
        ...actual,
        useRouter: () => {
          return {
            push: mockRoutePush,
          }
        },
      }
    })
  })

  beforeEach(async () => {
    await userQuery()
  })

  afterAll(() => {
    vi.clearAllMocks()
  })

  it('should match snapshot not logged', async () => {
    await noUserQuery()
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
    })
    await flushPromises()
    await nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot no data', async () => {
    await userNoDataQuery()
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
    })
    await flushPromises()
    await nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot logged', async () => {
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
    })
    await flushPromises()
    await nextTick()

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should list the correct amount of groups', async () => {
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
    })
    await flushPromises()
    await nextTick()

    expect(wrapper.findAll('.group-item').length).toBe(2)
  })

  it('should change to create group page after hitting create group', async () => {
    await router.replace({ path: '/groups' })
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
    })
    await flushPromises()
    await nextTick()

    await wrapper.find('.el-button').trigger('click')
    await nextTick()
    expect(mockRoutePush).toHaveBeenCalledWith('/group/create')
  })

  it('should navigate to group manage page', async () => {
    await router.replace({ path: '/groups' })
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
    })
    await flushPromises()
    await nextTick()

    wrapper.findAll('[data-test-key="manage-link"]').at(0).trigger('click')
    await nextTick()
    expect(mockRoutePush).toHaveBeenCalledWith({
      name: 'group',
      params: { groupIdOrSlug: mockGroups[0].urlSlug },
      query: { tab: 'settings' },
    })

    await router.replace({ path: '/groups' })
    wrapper.findAll('[data-test-key="manage-link"]').at(1).trigger('click')
    await nextTick()
    expect(mockRoutePush).toHaveBeenCalledWith({
      name: 'group',
      params: { groupIdOrSlug: mockGroups[1].urlSlug },
      query: { tab: 'members' },
    })
  })

  it('should navigate to browse dataset page', async () => {
    await router.replace({ path: '/groups' })
    const wrapper = mount(testHarness, {
      global: {
        plugins: [store, router],
        provide: {
          [DefaultApolloClient]: graphqlMocks,
        },
      },
    })
    await flushPromises()
    await nextTick()

    wrapper.findAll('[data-test-key="dataset-link"]').at(0).trigger('click')
    await nextTick()
    expect(mockRoutePush).toHaveBeenCalledWith({
      path: '/datasets',
      query: encodeParams({ group: mockGroups[0].id }),
    })

    await router.replace({ path: '/groups' })
    wrapper.findAll('[data-test-key="dataset-link"]').at(1).trigger('click')
    await nextTick()
    expect(mockRoutePush).toHaveBeenCalledWith({
      path: '/datasets',
      query: encodeParams({ group: mockGroups[1].id }),
    })
  })
})
