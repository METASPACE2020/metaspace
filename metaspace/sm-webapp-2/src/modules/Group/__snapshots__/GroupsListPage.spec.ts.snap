// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GroupsListItem > should match snapshot logged 1`] = `
"<div class=\\"groups-list-container\\">
  <div class=\\"groups-list-wrapper\\">
    <div class=\\"groups-list-header\\">
      <div class=\\"el-input el-input-group el-input-group--prepend group-name-filter tf-outer w-auto\\">
        <!-- input -->
        <!-- prepend slot -->
        <div class=\\"el-input-group__prepend\\">
          <div class=\\"mock-el-icon -mx-1\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
              <path fill=\\"currentColor\\" d=\\"m795.904 750.72 124.992 124.928a32 32 0 0 1-45.248 45.248L750.656 795.904a416 416 0 1 1 45.248-45.248zM480 832a352 352 0 1 0 0-704 352 352 0 0 0 0 704z\\"></path>
            </svg></div>
        </div>
        <div class=\\"el-input__wrapper\\" tabindex=\\"-1\\">
          <!-- prefix slot -->
          <!--v-if--><input class=\\"el-input__inner\\" type=\\"text\\" autocomplete=\\"off\\" tabindex=\\"0\\" placeholder=\\"Enter keywords\\" id=\\"el-id-replaced\\"><!-- suffix slot -->
          <!--v-if-->
        </div><!-- append slot -->
        <!--v-if-->
      </div><button ariadisabled=\\"false\\" type=\\"button\\" class=\\"el-button el-button--primary el-button--large group-list-create-btn\\">
        <!--v-if--><span class=\\"\\">Create group</span>
      </button>
    </div>
    <div class=\\"groups-list-content\\">
      <!---->
      <div class=\\"group-item\\">
        <div class=\\"group-item-info\\">
          <div class=\\"group-item-title-wrapper group-item-info-line\\">
            <div data-test-key=\\"group-link\\" class=\\"group-item-title\\">group1<span class=\\"group-item-short-name\\"> (g1)</span></div>
            <div data-v-0bc4b279=\\"\\" class=\\"mock-el-popover ml-1\\" trigger=\\"hover\\" placement=\\"top\\" popper-class=\\"min-w-0\\">
              <p data-v-0bc4b279=\\"\\" class=\\"leading-none m-0\\">Copy group id to clipboard</p>
            </div>
          </div>
          <div class=\\"group-item-info-line\\">
            <!----> 2 Members
          </div>
        </div>
        <div class=\\"group-item-actions\\">
          <div class=\\"flex items-center\\">
            <div class=\\"mock-el-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
                <path fill=\\"currentColor\\" d=\\"M96 896a32 32 0 0 1-32-32V160a32 32 0 0 1 32-32h832a32 32 0 0 1 32 32v704a32 32 0 0 1-32 32H96zm315.52-228.48-68.928-68.928a32 32 0 0 0-45.248 0L128 768.064h778.688l-242.112-290.56a32 32 0 0 0-49.216 0L458.752 665.408a32 32 0 0 1-47.232 2.112zM256 384a96 96 0 1 0 192.064-.064A96 96 0 0 0 256 384z\\"></path>
              </svg></div>
            <div class=\\"link ml-1\\" data-test-key=\\"dataset-link\\">Browse datasets</div>
          </div>
          <div class=\\"flex items-center\\">
            <div class=\\"mock-el-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
                <path fill=\\"currentColor\\" d=\\"m199.04 672.64 193.984 112 224-387.968-193.92-112-224 388.032zm-23.872 60.16 32.896 148.288 144.896-45.696L175.168 732.8zM455.04 229.248l193.92 112 56.704-98.112-193.984-112-56.64 98.112zM104.32 708.8l384-665.024 304.768 175.936L409.152 884.8h.064l-248.448 78.336L104.32 708.8zm384 254.272v-64h448v64h-448z\\"></path>
              </svg></div>
            <div class=\\"link ml-1\\" data-test-key=\\"manage-link\\">Manage group</div>
          </div>
        </div>
      </div>
      <div class=\\"group-item\\">
        <div class=\\"group-item-info\\">
          <div class=\\"group-item-title-wrapper group-item-info-line\\">
            <div data-test-key=\\"group-link\\" class=\\"group-item-title\\">group2<span class=\\"group-item-short-name\\"> (g2)</span></div>
            <div data-v-0bc4b279=\\"\\" class=\\"mock-el-popover ml-1\\" trigger=\\"hover\\" placement=\\"top\\" popper-class=\\"min-w-0\\">
              <p data-v-0bc4b279=\\"\\" class=\\"leading-none m-0\\">Copy group id to clipboard</p>
            </div>
          </div>
          <div class=\\"group-item-info-line\\">
            <!----> 1 Member
          </div>
        </div>
        <div class=\\"group-item-actions\\">
          <div class=\\"flex items-center\\">
            <div class=\\"mock-el-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
                <path fill=\\"currentColor\\" d=\\"M96 896a32 32 0 0 1-32-32V160a32 32 0 0 1 32-32h832a32 32 0 0 1 32 32v704a32 32 0 0 1-32 32H96zm315.52-228.48-68.928-68.928a32 32 0 0 0-45.248 0L128 768.064h778.688l-242.112-290.56a32 32 0 0 0-49.216 0L458.752 665.408a32 32 0 0 1-47.232 2.112zM256 384a96 96 0 1 0 192.064-.064A96 96 0 0 0 256 384z\\"></path>
              </svg></div>
            <div class=\\"link ml-1\\" data-test-key=\\"dataset-link\\">Browse datasets</div>
          </div>
          <div class=\\"flex items-center\\">
            <div class=\\"mock-el-icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
                <path fill=\\"currentColor\\" d=\\"m199.04 672.64 193.984 112 224-387.968-193.92-112-224 388.032zm-23.872 60.16 32.896 148.288 144.896-45.696L175.168 732.8zM455.04 229.248l193.92 112 56.704-98.112-193.984-112-56.64 98.112zM104.32 708.8l384-665.024 304.768 175.936L409.152 884.8h.064l-248.448 78.336L104.32 708.8zm384 254.272v-64h448v64h-448z\\"></path>
              </svg></div>
            <div class=\\"link ml-1\\" data-test-key=\\"manage-link\\">Browse group</div>
          </div>
        </div>
      </div>
      <!---->
    </div>
  </div>
</div>"
`;

exports[`GroupsListItem > should match snapshot no data 1`] = `
"<div class=\\"groups-list-container\\">
  <div class=\\"groups-list-wrapper\\">
    <div class=\\"groups-list-header\\">
      <div class=\\"el-input el-input-group el-input-group--prepend group-name-filter tf-outer w-auto\\">
        <!-- input -->
        <!-- prepend slot -->
        <div class=\\"el-input-group__prepend\\">
          <div class=\\"mock-el-icon -mx-1\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\">
              <path fill=\\"currentColor\\" d=\\"m795.904 750.72 124.992 124.928a32 32 0 0 1-45.248 45.248L750.656 795.904a416 416 0 1 1 45.248-45.248zM480 832a352 352 0 1 0 0-704 352 352 0 0 0 0 704z\\"></path>
            </svg></div>
        </div>
        <div class=\\"el-input__wrapper\\" tabindex=\\"-1\\">
          <!-- prefix slot -->
          <!--v-if--><input class=\\"el-input__inner\\" type=\\"text\\" autocomplete=\\"off\\" tabindex=\\"0\\" placeholder=\\"Enter keywords\\" id=\\"el-id-replaced\\"><!-- suffix slot -->
          <!--v-if-->
        </div><!-- append slot -->
        <!--v-if-->
      </div><button ariadisabled=\\"false\\" type=\\"button\\" class=\\"el-button el-button--primary el-button--large group-list-create-btn\\">
        <!--v-if--><span class=\\"\\">Create group</span>
      </button>
    </div>
    <div class=\\"groups-list-content\\">
      <p>Group not found!</p>
      <!---->
      <!---->
    </div>
  </div>
</div>"
`;

exports[`GroupsListItem > should match snapshot not logged 1`] = `
"<div class=\\"groups-list-container\\">
  <div class=\\"groups-list-wrapper\\">
    <p class=\\"font-normal text-center\\">Create an account/login, to create groups and make your work reach more people!</p>
  </div>
</div>"
`;
